Add reqmgr2ms-pileup service

vkuznet opened the pull request:
Fixes #11423 

#### Status
in development

#### Description
First draft for MSPileup service

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
- [Dockerfile](https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/reqmgr2ms-pileup/Dockerfile)
- [apache frontend rules](https://github.com/dmwm/deployment/pull/1230)
- reqmgr2ms-pileup configuration draft can be found in this [config.py](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/test/reqmgr2ms-pileup/config.py)
- [reqmgr2ms-pileup.yaml](https://github.com/dmwm/CMSKubernetes/blob/master/kubernetes/cmsweb/services/reqmgr2ms-pileup.yaml) k8s manifest file to deploy service to k8s cluster
- reqmgr2ms-pileup configuration draft can be found in gitlab [services_configs/reqmgr2ms-pileup/config.py](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/test/reqmgr2ms-pileup/config.py) repo
I used port 8249 as new port number for this service

The service skeleton is also tested with local setup using curl. see https://github.com/dmwm/WMCore/pull/11430#issuecomment-1376324420

Additional changes made by @arooshap :
- For the cmsweb cluster: [services_config/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/commit/ce8c18838e7dcd22162aaf61834ea48c64a48dd7).
- For the preprod cluster: [services_config/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/commit/fa4cd17cd67e8d26b5b46916fab145e4171ca288).
- and for the test cluster: [services_config/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/commit/bd07af9c422fc2dd3cd45428d7fec7eb1ca3badf).

#### External dependencies / deployment changes

vkuznet committed:
Add reqmgr2ms-pileup

vkuznet commented:
@amaltaro , @todor-ivanov I made initial attempt to create MSPileup service. The changes are spread across multiple projects and I'll summarize them here:
- initial code base for MSPileup service, this PR
- changes to setup_dependencies.py, this PR
- changes to CI/CD, this PR
- [Dockerfile](https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/reqmgr2ms-pileup/Dockerfile)
- [frontend redirect rules](https://github.com/dmwm/deployment/pull/1229) in deployment repo
- reqmgr2ms-pileup configuration draft can be found in gitlab [services_configs/reqmgr2ms-pileup/config.py](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/test/reqmgr2ms-pileup/config.py) repo
  - I used port 8249 as new port number for this service
- [reqmgr2ms-pileup.yaml](https://github.com/dmwm/CMSKubernetes/blob/master/kubernetes/cmsweb/services/reqmgr2ms-pileup.yaml) k8s manifest file to deploy service to k8s cluster

I saw that pycurl_manager failed during unit tests which seems completely irrelevant here and rather an issue with access to cms-gwmsmon.cern.ch service, see [here](https://cmssdt.cern.ch/dmwm-jenkins/job/DMWM-WMCore-PR-test/13865/testReport/junit/WMCore_t.Services_t.pycurl_manager_t/PyCurlManager/testSingle/). I suggest that someone decide if we need to use this service in pycurl tests anymore but I doubt it has anything to do with this PR.

Said that, I need your initial review to proceed. If anything is missing or should be added feel free to point me to it. Meanwhile I tested new service in local setup and was able to get proper values from`/ms-pileup/status` and `/ms-pileup/info` end-points:
```
curl -v http://localhost:8249/ms-pileup/data/status
...
< HTTP/1.1 200 OK
...
{"result": [
 {
  "wmcore_version": "2.1.6rc5",
  "microservice_version": "2.1.6rc5",
  "microservice": "MSManager",
  "status": "OK"
}]}

curl -v http://localhost:8249/ms-pileup/data/info
...
< HTTP/1.1 200 OK
...
{"result": [
 {
  "wmcore_version": "2.1.6rc5",
  "microservice_version": "2.1.6rc5",
  "microservice": "MSManager",
  "request": null,
  "transferDoc": null
}]}
```

amaltaro commented on .github/workflows/pypi_build_and_images.yaml:
```diff
@@ -27,7 +27,7 @@ jobs:
       matrix:
         target: [wmagent, wmagent-devtools, wmcore, reqmon, reqmgr2, global-workqueue, acdcserver, reqmgr2ms-unmerged,
-                 reqmgr2ms-output, reqmgr2ms-rulecleaner, reqmgr2ms-transferor, reqmgr2ms-monitor]
+                 reqmgr2ms-output reqmgr2ms-pileup, reqmgr2ms-rulecleaner, reqmgr2ms-transferor, reqmgr2ms-monitor]
```
Missing comma here.

vkuznet replied:
done

amaltaro reviewed (changes requested):
@vkuznet all these changes are looking pretty good. I made a comment along the code that needs fixing.

In addition to that:
* on the Dockerfile, you have a typo in there `RUN pip install reqmgr2ms-pielup==$TAG`
* for the config.py, in services_config, I would suggest to start it with the bare minimum configuration at the moment and add attributes as we need them. I can get back to you with a candidate configuration soon.

Can you please also make sure that your comment in the middle of this PR gets incorporated to the very first comment in the PR?

vkuznet commented:
Alan, thanks for spotting missing comma and typo. I fixed both.

vkuznet commented:
Alan, I squashed the changes, updated PR description, and I think this PR is ready to go. Feel free to review again and merge. The confit file can be updated separately via gitlab PR and you may submit your version over there. Once this PR is merged I can continue on separate PRs for other (implementation) parts of new service.

amaltaro reviewed (approved):
It looks all good, thanks!

amaltaro commented:
PS.: I am removing this PR from the project because the project is supposed to be used only for GH issues. PR labels usually start with "PR:" as well, so I am removing those that are unneeded.
